cypress/plugins/nodeManager/commands.ts

```diff
@@ -0,0 +1,51 @@
+import type {
+  ConnectNodesOptions,
+  OnboardNodeOptions,
+  OnboardedNode,
+  StartNodeOptions,
+} from "./shared";
+
+import { Commands } from "./shared";
+
+const startNode = (options: StartNodeOptions): void => {
+  cy.task(Commands.StartNode, options);
+};
+
+const onboardNode = (options: OnboardNodeOptions): void => {
+  cy.task(Commands.OnboardNode, options);
+};
+
+const connectNodes = (options: ConnectNodesOptions): void => {
+  cy.task(Commands.ConnectNodes, options);
+};
+
+const stopAllNodes = (): void => {
+  cy.task(Commands.StopAllNodes);
+};
+
+export const withTwoConnectedNodes = (
+  callback: (node0: OnboardedNode, node1: OnboardedNode) => void
+): void => {
+  startNode({ id: 17000 });
+  onboardNode({ id: 17000 });
+
+  startNode({ id: 18000 });
+  onboardNode({ id: 18000 });
```

Comment:
We could return `OnboardedNode` from this function. This would eliminate the need for `Commands.GetOnboardedNodes` (which is susceptible to exceptions if the number of onboarded nodes does not match the number we expect`.

Reply:
I don't want to spend more time refactoring before we know a bit more about what real life test-cases will look like.
While a callback wrapping approach might be more type-safe, it might make testing non-linear process lifecycles more obfuscated.

I suggest we write a couple of tests and do a follow-up if necessary.